### ISSUE TITLE
Disabling clippy lint warning for specific lines

### ISSUE THREAD

**Issue by SirWindfield** [rust-lang/rust-clippy#3904]
Hey!

I am currently adding clippy to a project and want to disable to lines triggering the `clippy::or_fun_call`. I can't actually change them because the underlying value gets moved earlier on and borrowing it within a closure doesn't work in my case.

I've read that I can disable the entire lint my using `#![allow(clippy::or_fun_call)]` but that would ignore it for the whole project.
Adding `#[allow(clippy::or_fun_call)]` infront of the line as I did in the code below doesn't work too (someone mentioned this idea on stackoverflow). 

The underlying line is the fourth one within this block:
```rust
let discovery_stream = discovery(
        &handle,
        ConnectConfig {
            #[allow(clippy::or_fun_call)] name: matches.opt_str("device_name").unwrap_or(config.device_name.clone()),
            device_type: DeviceType::default(),
            volume: mixer().volume(),
            linear_volume,
        },
        device_id,
        0,
    ).unwrap();
```

**Comment by Manishearth** [comment]
You have to do i on the enclosing function.

(Putting it on the `let` statement _may_ work)

**Comment by SirWindfield** [comment]
For future reference, extracting the value into a `let` statement did work.
